Kernel/Modules/AgentTicketCompose.pm

```diff
@@ -86,19 +86,19 @@ sub new {
             }
         }
 
-        push $Self->{DynamicField}->@*, $DynamicField;
+        $Self->{DynamicField}{ $DynamicField->{Name} } = $DynamicField;
     }
 
     # align sysconfig and ticket mask data II
     for my $DynamicFieldName ( keys $Definition->{DynamicFields}->%* ) {
-        push $Self->{DynamicField}->@*, $DynamicFieldObject->DynamicFieldGet(
+        $Self->{DynamicField}{$DynamicFieldName} = $DynamicFieldObject->DynamicFieldGet(
             Name => $DynamicFieldName,
         );
 
         my $Parameters = $Definition->{DynamicFields}{$DynamicFieldName} // {};
 
         for my $Attribute ( keys $Parameters->%* ) {
-            $Self->{DynamicField}[-1]{$Attribute} = $Parameters->{$Attribute};
+            $Self->{DynamicField}{$DynamicFieldName}{$Attribute} = $Parameters->{$Attribute};
         }
```

Comment:
Please align the code more with the main masks, AgentTicketActionCommon would be the most general one, even though some things, like the differentiation of ticket and article dynamic fields are not necessary here, of course.

In the new method therefor please also define $Self->{FieldMethods} (only containing the method for the next state), as well as $Self->{InternalDependancy} (as empty hash).